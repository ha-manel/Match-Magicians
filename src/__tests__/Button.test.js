import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Button from '../components/button';
import '@testing-library/jest-dom';

describe('Testing the button component', () => {
  const calculateResult = jest.fn();

  it('Button component is rendered', () => {
    const button = renderer
      .create(<Button eventHandler={calculateResult} value="AC" />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Test button click', () => {
    render(<Button eventHandler={calculateResult} value="AC" />);
    fireEvent.click(screen.getByText('AC'));
    fireEvent.click(screen.getByText('AC'));
    fireEvent.click(screen.getByText('AC'));
    expect(calculateResult).toHaveBeenCalledTimes(3);
  });
});
