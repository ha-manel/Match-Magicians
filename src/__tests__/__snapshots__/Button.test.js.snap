// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the button component Button component is rendered 1`] = `
<button
  className=""
  onClick={[MockFunction]}
  type="button"
>
  AC
</button>
`;
